src/helpers/snapshot.ts

```diff
@@ -0,0 +1,168 @@
+import { getBlockNumber } from '@snapshot-labs/snapshot.js/src/utils/web3';
+import { ipfsGet, getScores } from '@snapshot-labs/snapshot.js/src/utils';
+import { formatProposal, formatProposals } from '@/helpers/utils';
+import { getProfiles } from '@/helpers/profile';
+import getProvider from '@snapshot-labs/snapshot.js/src/utils/provider';
+import gateways from '@snapshot-labs/snapshot.js/src/gateways.json';
+import client from '@/helpers/client';
+
+const gateway = process.env.VUE_APP_IPFS_GATEWAY || gateways[0];
+
+let proposalRes;
+
+export async function getProposal(id) {
+  try {
+    console.time('getProposal.data');
+    proposalRes = await Promise.all([ipfsGet(gateway, id)]);
+    console.timeEnd('getProposal.data');
+    let [proposal]: any = proposalRes;
+    proposal = formatProposal(proposal);
+    proposal.ipfsHash = id;
+
+    return { proposal };
+  } catch (e) {
+    console.log(e);
+    return e;
+  }
+}
+
+export async function getResults(space, id) {
+  try {
+    const provider = getProvider(space.network);
+    console.time('getProposal.results');
+    const response = await Promise.all([
+      client.request(`${space.key}/proposal/${id}`),
+      getBlockNumber(provider)
+    ]);
+    console.timeEnd('getProposal.results');
+    const [, blockNumber] = response;
+    let [votes]: any = response;
+    let [proposal]: any = proposalRes;
```

Comment:
This is not a good way to get the proposal data, the function should be self contained, the proposal data can be passed in the function params. You request getProposal and getResults in same time there is no guarantee that the `proposalRes` will be available here.